### PR TITLE
Add GitHub organization data to users document on registration

### PR DESCRIPTION
**Pull request by owap**
*(Please merge pull request #5 first to see an accurate diff)*

This pull request adds an `organizations` field to the `User` model, and queries the GitHub API when registering with our application to pull back organizations to which the user belongs.

After registration, the user data in MongoDB looks like this (see `organizations` section near the bottom):

```bson
{ 
 "username" : "owap",
 "displayName" : "Andrew J. Pierce",
 "provider" : "github",
 "providerData" : { "accessToken" : "fe18edabd4f892faa84b113075da152691b93629",
 "plan" : { "private_repos" : 0,
 "collaborators" : 0,
 "space" : 976562499,
 "name" : "free" },
 "collaborators" : 0,
 "disk_usage" : 5682,
 "owned_private_repos" : 0,
 "total_private_repos" : 10,
 "private_gists" : 0,
 "updated_at" : "2015-06-02T12:31:29Z",
 "created_at" : "2014-08-19T21:37:58Z",
 "following" : 2,
 "followers" : 2,
 "public_gists" : 0,
 "public_repos" : 6,
 "bio" : null,
 "hireable" : false,
 "email" : "",
 "location" : "Herndon, VA",
 "blog" : "https://github.com/ajpierce",
 "company" : "OpenWhere",
 "name" : "Andrew J. Pierce",
 "site_admin" : false,
 "type" : "User",
 "received_events_url" : "https://api.github.com/users/owap/received_events",
 "events_url" : "https://api.github.com/users/owap/events{/privacy}",
 "repos_url" : "https://api.github.com/users/owap/repos",
 "organizations_url" : "https://api.github.com/users/owap/orgs",
 "subscriptions_url" : "https://api.github.com/users/owap/subscriptions",
 "starred_url" : "https://api.github.com/users/owap/starred{/owner}{/repo}",
 "gists_url" : "https://api.github.com/users/owap/gists{/gist_id}",
 "following_url" : "https://api.github.com/users/owap/following{/other_user}",
 "followers_url" : "https://api.github.com/users/owap/followers",
 "html_url" : "https://github.com/owap",
 "url" : "https://api.github.com/users/owap",
 "gravatar_id" : "",
 "avatar_url" : "https://avatars.githubusercontent.com/u/8496383?v=3",
 "id" : 8496383,
 "login" : "owap" },
 "created" : ISODate("2015-06-05T16:49:50.581Z"),
 "roles" : [ "user" ],
 "organizations" : [ 
    { "description" : null,
     "avatar_url" : "https://avatars.githubusercontent.com/u/271861?v=3",
     "public_members_url" : "https://api.github.com/orgs/kazzaz/public_members{/member}",
     "members_url" : "https://api.github.com/orgs/kazzaz/members{/member}",
     "events_url" : "https://api.github.com/orgs/kazzaz/events",
     "repos_url" : "https://api.github.com/orgs/kazzaz/repos",
     "url" : "https://api.github.com/orgs/kazzaz",
     "id" : 271861,
     "login" : "kazzaz" },
     { "description" : "",
     "avatar_url" : "https://avatars.githubusercontent.com/u/7431080?v=3",
     "public_members_url" : "https://api.github.com/orgs/OpenWhere/public_members{/member}",
     "members_url" : "https://api.github.com/orgs/OpenWhere/members{/member}",
     "events_url" : "https://api.github.com/orgs/OpenWhere/events",
     "repos_url" : "https://api.github.com/orgs/OpenWhere/repos",
     "url" : "https://api.github.com/orgs/OpenWhere",
     "id" : 7431080,
     "login" : "OpenWhere" } 
 ],
 "password" : "",
 "email" : "",
 "__v" : 0
}
```

### CODE REVIEW

File: app/controllers/users/users.authentication.server.controller.js [hunk 1]
```diff
@@ -124,27 +126,44 @@ exports.saveOAuthUserProfile = function(req, providerUserProfile, done) {
...
-                            email: providerUserProfile.email,
-                            provider: providerUserProfile.provider,
-                            providerData: providerUserProfile.providerData
-                        });
-
-                        // And save the user
-                        user.save(function(err) {
-                            return done(err, user);
+                // Function for saving user information in the database
+                var createUserCallback = function(organizations){
```
Comment (owaxh) [inline app/controllers/users/users.authentication.server.controller.js]: this was a bit hard to follow for me.   Would it be better to make the createUserCallabck a helper function outside of exports.saveOAuthUserProfile ?  The logic follow maybe easier to read

Comment (owap) [inline app/controllers/users/users.authentication.server.controller.js]: I'm all about readability! I started refactoring this to a helper function outside of `saveOAuthUserProfile`, but it started turning into something more complex.

Specifically, refactoring the function requires us to pass in variables that were previously in scope, including `providerUserProfile`, `user`, `err`, and the `done` callback.  As it stands right now, we only pass in `organizations` or nothing, and it saves the user.  We *can* pass these variables to an outside function, but I think leveraging variables provided by keeping the function as a locally-defined closure is a more idiomatic use of Javascript.

Comment (owaxh) [inline app/controllers/users/users.authentication.server.controller.js]: what about passing "this"

**Comment by owap** [comment]
Pulled `createUserCallback` out into a top-level function to enhance readability as per comments

File: app/controllers/users/users.authentication.server.controller.js (hunk 1 above)
Comment (owap) [inline app/controllers/users/users.authentication.server.controller.js]: The other variables are not assigned to `this`, so passing the scope to an outside function wouldn't help us in terms of accessing the variables.  I've refactored this to pass the other variables, and re-tested user registration -- works just like before

**Comment by owaxh** [comment]
lgtm
